python/cudf/cudf/_lib/column.pyx

```diff
@@ -106,6 +120,18 @@ cdef class Column:
             start = self.offset * self.dtype.itemsize
             end = start + self.size * self.dtype.itemsize
             self._data = self.base_data[start:end]
+            # special case of string column
```

Comment:
At this point, is there anywhere that we're attempting to access the `.data` attribute for a strings column?

If not, I'd prefer we just raise `NotImplementedError` when someone attempts to do that, instead of setting `.data` to be the same as `.base_data` for strings column specially. It seems confusing.

Reply:
`.data` is used by`ColumnBase.serialize()` on all type of columns.
https://github.com/rapidsai/cudf/blob/d311a6806d12f6f325880b6a9674c486336a7dc6/python/cudf/cudf/core/column/column.py#L1213

If setting `.data` to be same as `.base_data` is the only concern, we could choose to copy it always.

Reply:
Moved this to string.py. Simplified.